app/src/main/database/index.ts

```diff
@@ -297,6 +359,108 @@ export class DB {
     })(metadata);
   }
 
+  /**
+   * Update item content (plaintext for messages, filename for files)
+   * @param itemId - UUID of the item to update
+   * @param options - Update options
+   * @param options.plaintext - The decrypted plaintext content (only allowed for messages)
+   * @param options.filename - The original filename (only allowed for files)
+   */
+  updateItem(
+    itemId: string,
+    options: { plaintext?: string; filename?: string },
+  ) {
+    if (!this.db) {
+      throw new Error("Database not initialized");
+    }
+
+    // Get the item to check its type
+    const items = this.getItems([itemId]);
+    if (items.length === 0) {
+      throw new Error(`Item with ID ${itemId} not found`);
+    }
+
+    const item = items[0];
+    const { plaintext, filename } = options;
+
+    // Validate update based on item type
+    if (
+      plaintext !== undefined &&
+      item.data.kind !== "message" &&
+      item.data.kind !== "reply"
+    ) {
+      throw new Error(
+        `Cannot update plaintext for item of kind "${item.data.kind}". Only "message" and "reply" items can have plaintext updated.`,
+      );
+    }
+
+    if (filename !== undefined && item.data.kind !== "file") {
+      throw new Error(
+        `Cannot update filename for item of kind "${item.data.kind}". Only "file" items can have filename updated.`,
+      );
+    }
+
+    // Use appropriate prepared statement based on what's being updated
+    if (plaintext !== undefined && filename !== undefined) {
+      this.updateItemBoth.run({ id: itemId, plaintext, filename });
+    } else if (plaintext !== undefined) {
+      this.updateItemPlaintext.run({ id: itemId, plaintext });
+    } else if (filename !== undefined) {
+      this.updateItemFilename.run({ id: itemId, filename });
+    }
+  }
+
+  /**
+   * Get items by their UUIDs
+   * @param itemIds - Array of item UUIDs to retrieve
+   * @returns Array of Item objects
+   */
+  getItems(itemIds: string[]): Item[] {
+    if (!this.db) {
+      throw new Error("Database not initialized");
+    }
+
+    if (itemIds.length === 0) {
+      return [];
+    }
+
+    const items: Item[] = [];
+    for (const itemId of itemIds) {
+      const row = this.selectItemById.get(itemId);
+      if (row) {
+        const data = JSON.parse(row.data) as ItemMetadata;
+        items.push({
+          uuid: row.uuid,
+          data,
+          plaintext: row.plaintext || undefined,
+          filename: row.filename || undefined,
+        });
+      }
+    }
+
+    return items;
+  }
+
+  /**
+   * Get UUIDs of all message and reply items that don't have plaintext
+   * @returns Array of item UUIDs that need decryption
+   */
+  getUndecryptedMessageIds(): string[] {
```

Comment:
I think we can now remove this (+ `selectUndecryptedMessages`) since we no longer need to manually queue the undecrypted messages!